src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1199,6 +1199,12 @@ ImmutableArray<AttributeData> IMethodSymbol.GetReturnTypeAttributes()
         /// </summary>
         internal virtual void AddSynthesizedReturnTypeAttributes(ref ArrayBuilder<SynthesizedAttributeData> attributes)
         {
+            if (this.ReturnsByRefReadonly)
+            {
+                // PROTOTYPE(readonlyRefs) it is optional now as it will be generated in the next PR
+                var constructor = WellKnownMember.System_Runtime_CompilerServices_ReadOnlyAttribute__ctor;
+                AddSynthesizedAttribute(ref attributes, this.DeclaringCompilation.TrySynthesizeAttribute(constructor, isOptionalUse: true));
```

Comment:
This attribute is synthesized in multiple places. Consider adding a `CSharpCompilation.SynthesizeReadOnlyAttribute()` similar to handling of other attributes.